File: src/base/framework/python/ossie/component.py
from ossie.cf import CF
from resource import Resource
import containers
from omniORB import CORBA

class Component(Resource):
    def __init__(self, identifier, execparams, propertydefs=(), loggerName=None):
        super(Component,self).__init__(identifier, execparams, propertydefs, loggerName)
        self._app = None
        
    def getApplication(self):
        return self._app
    
    def setAdditionalParameters(self, softwareProfile, application_registrar_ior):
        super(Component,self).setAdditionalParameters(softwareProfile, application_registrar_ior)
        orb = CORBA.ORB_init()
        obj = orb.string_to_object(application_registrar_ior)
        applicationRegistrar = obj._narrow(CF.ApplicationRegistrar)
        if applicationRegistrar != None:
            self._app = containers.ApplicationContainer(applicationRegistrar._get_app())
    

File: src/base/framework/java/ossie/src/org/ossie/component/Component.java
package org.ossie.component;

import org.omg.CosNaming.NamingContext;
import org.omg.CosNaming.NamingContextHelper;

import org.omg.CORBA.ORB;
import CF.ResourceOperations;
import CF.Application;
import CF.ApplicationRegistrar;
import CF.ApplicationRegistrarHelper;
import CF.DomainManager;
import org.ossie.component.ThreadedResource;
import org.ossie.component.ThreadedComponent;
import org.ossie.redhawk.ApplicationContainer;

public abstract class Component extends ThreadedResource implements ResourceOperations,ThreadedComponent {
    protected ApplicationContainer _app;
    
    public Component() {
        super();
        this._app = null;
    }
    
    public ApplicationContainer getApplication() {
        return this._app;
    }
    
    public void setAdditionalParameters(String ApplicationRegistrarIOR) {
        super.setAdditionalParameters(ApplicationRegistrarIOR);
        final org.omg.CORBA.Object obj = this.orb.string_to_object(ApplicationRegistrarIOR);
        ApplicationRegistrar appReg = null;
        try {
            appReg = ApplicationRegistrarHelper.narrow(obj);
        } catch (Exception e) {}
        if (appReg!=null) {
            this._app = new ApplicationContainer(appReg.app());
        }
    }
};
